forem/forem
# "Reply to" and "From" Email addresses for SMTP Configurations

## What type of PR is this? (check all applicable)


## Description

This PR introduces the ability to add a `reply_to_email_address` and a `from_email_address` for custom SMTP Settings.
When you set these email addresses it will then send all mails from the correct email address, and have the correct reply-to email address as well. 

It will also set some default values based on the "DEFAULT_EMAIL" environment variable that will be filled out both during a self-hosted and Forem Cloud setup.

I wan't sure how to update the reply-to on Devise Mailers and so I haven't tackled that as yet and will do it as a separate PR.

Demo Video
https://www.loom.com/share/cdb278917765402189e38f908bb03608

## Related Tickets & Documents
https://github.com/forem/forem/issues/16180

## QA Instructions, Screenshots, Recordings

Without Custom SMTP Settings:
1. Set your DEFAULT_EMAIL variable in your .env, and restart your server. Make sure that http://localhost:3000/admin/customization/config#smtp-section has no username, password and address.
2. Now go to http://localhost:3000/rails/mailers/notify_mailer/account_deleted_email, you should see the from and reply to addresses match your DEFAULT_EMAIL env variable.

With Custom SMTP Settings:
1. Go to http://localhost:3000/admin/customization/config#smtp-section and add a username, password and address.
2. Update the from and reply to email address as well.
3. Now go to http://localhost:3000/rails/mailers/notify_mailer/account_deleted_email, you should see the from and reply to addresses match your updated SMTP Values. 

### UI accessibility concerns?
none.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/CuHO2q0lorhBieOEMp/giphy.gif)
